File: Mar-18/26-03-18/c++/01-Basic-of-Stack.cpp
#include<bits/stdc++.h>

using namespace std;

int main()
{
    int q;
    cin >> q;
    stack<int> st;
    while(q--)
    {
        int x, y;
        cin >> x;
        if(x == 1)
        {
            if(st.empty())
                cout << "No Food\n";
            else
            {
                cout << st.top() << endl;
                st.pop();
            }
        }
        else
        {
            cin >> y;
            st.push(y);
        }
    }
}

File: Mar-18/26-03-18/java/01-Basic-of-Stack.java

import java.util.*;

class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner input = new Scanner(System.in);
        int q = input.nextInt();
        
        Stack<Integer> st = new Stack<Integer>();
        
        while(q-- > 0)
        {
            int x, y;
            x = input.nextInt();
            if(x == 1)
            {
                if(st.empty())
                    System.out.println("No Food");
                else
                {
                    System.out.println(st.peek());
                    st.pop();
                }
            }
            else
            {
                y = input.nextInt();
                st.push(y);
            }
        }

    }
}
